shotover-proxy/src/transforms/cassandra/sink_cluster/token_map.rs

```diff
@@ -0,0 +1,109 @@
+use crate::transforms::cassandra::sink_cluster::CassandraNode;
+use cassandra_protocol::token::Murmur3Token;
+use std::collections::BTreeMap;
+use uuid::Uuid;
+
+#[derive(Debug, Clone)]
+pub struct TokenMap {
+    token_ring: BTreeMap<Murmur3Token, Uuid>,
+}
+
+impl TokenMap {
+    pub fn new(nodes: &[CassandraNode]) -> Self {
+        TokenMap {
+            token_ring: nodes
+                .iter()
+                .flat_map(|node| node.tokens.iter().map(move |token| (*token, node.host_id)))
+                .collect(),
+        }
+    }
+
+    /// Returns local nodes starting at given token and going in the direction of replicas.
```

Comment:
I would like to stop using the phrase "local nodes" (local_nodes has been renamed in https://github.com/shotover/shotover-proxy/pull/831)

Reply:
Yeah I did think it was confusing, espc because Cassandra has it's own concept of local nodes.